azure/deploy-to-azure/azure-integration/azuredeploy.json

```diff
@@ -198,7 +201,105 @@
+            "condition": "[not(empty(parameters('hostFilters')))]",
+            "type": "Microsoft.Resources/deploymentScripts",
+            "apiVersion": "2020-10-01",
+            "name": "[concat('datadog-host-filter-script-', parameters('newguid'))]",
+            "location": "[parameters('location')]",
+            "kind": "AzurePowerShell",
+            "dependsOn": [
+                "[resourceId('Microsoft.Resources/deploymentScripts', concat('datadog-integration-script-', parameters('newguid')))]"
+            ],
+            "properties": {
+                "environmentVariables": [
+                    {
+                        "name": "datadogApplicationKey",
+                        "secureValue": "[parameters('datadogApplicationKey')]"
+                    },
+                    {
+                        "name": "datadogApiKey",
+                        "secureValue": "[parameters('datadogApiKey')]"
+                    },
+                    {
+                        "name": "datadogSite",
+                        "value": "[parameters('datadogSite')]"
+                    },
+                    {
+                        "name": "hostFilters",
+                        "value": "[parameters('hostFilters')]"
+                    },
+                    {
+                        "name": "namespace",
+                        "value": "host"
+                    },
+                    {
+                        "name": "clientId",
+                        "value": "[parameters('servicePrincipalClientId')]"
+                    },
+                    {
+                        "name": "tenantName",
+                        "value": "[subscription().tenantId]"
+                    }
+                ],
+                "retentionInterval": "PT1H",
+                "timeout": "PT2M",
+                "containerSettings": {},
+                "cleanupPreference": "Always",
+                "azPowerShellVersion": "8.1",
+                "scriptContent": "Invoke-WebRequest -SkipCertificateCheck -ContentType \"application/json\" -Method Post -Uri \"https://api.$Env:datadogSite/api/v1/integration/azure/filtering\" -Headers @{\"Accept\"=\"application/json\"; \"DD-APPLICATION-KEY\"=$Env:datadogApplicationKey; \"DD-API-KEY\"=$Env:datadogApiKey} -Body (@{\"client_id\"=$Env:clientId; \"tenant_id\"=$Env:tenantName; \"namespace\"=$Env:namespace; \"filters\"=$Env:hostFilters} | ConvertTo-Json )"
+            }
+        },
+        {
+            "condition": "[not(empty(parameters('appFilters')))]",
+            "type": "Microsoft.Resources/deploymentScripts",
+            "apiVersion": "2020-10-01",
+            "name": "[concat('datadog-app-filter-script-', parameters('newguid'))]",
+            "location": "[parameters('location')]",
+            "kind": "AzurePowerShell",
+            "dependsOn": [
+                "[resourceId('Microsoft.Resources/deploymentScripts', concat('datadog-integration-script-', parameters('newguid')))]"
```

Comment:
If I'm understanding, based on this, both of the filters upserts can be submitted concurrently. We should test to make sure this consistently works and doesn't cause some sort of concurrency-based failure in the backend. If we see issues, we should just make one of the filter upserts depend on the other

Reply:
I had tested this a bunch of times and it works fine keeping the above in mind.

Some thoughts:
- Flask web server can handle multiple requests though it is not very scalable. We would have max 2 requests. It queues them if it is configured as single threaded.
-  the primary key for the `azure_namespace_filters` is `(org_id, tenant_id, client_id, namespace)` so there should not be any issues there

(Also, it's very unlikely that the 2 filter requests would be sent at the same time - given that 2 temporary containers need to be spun up to run these scripts)

Reply:
I'm convinced, thanks!